scripts/setup-adapters.sh

```diff
@@ -16,6 +16,114 @@
 # Propagate errors and improve debugging.
 set -eufx -o pipefail
 
+function install_gcs-sdk-cpp {
+  #install google-cloud-cpp dependencies (ubuntu 20.04)
+  echo " Installing apt dependencies..."
+  export DEBIAN_FRONTEND=noninteractive
```

Comment:
> I found the discussion here https://issues.apache.org/jira/browse/ARROW-1231 helpful. @ecopty It will be nice to include the emulator along with the tests in this PR for completeness. You can probably keep it as a separate commit to help simplify the review.

sure @majetideepak, thank you for your reference, I was planning to add testing as a next step, but needed to do the research for emulator first (thanks!), in the meantime I appreciate your code review comments and I will attend to them.